src/StudentsController.ts

```diff
@@ -300,7 +302,20 @@ export class StudentsController {
 
         const schoolLogoPNGPath = path.join(this.assetsLocation, "school_logo.png");
         const schoolLogoJPGPath = path.join(this.assetsLocation, "school_logo.jpg");
-        if (fs.existsSync(schoolLogoPNGPath)) {
+        if (data.schoolLogo) {
+            const bytes = Buffer.from(data.schoolLogo, "base64");
+            const matcher = bytes.toString("hex", 0, 4);
+
+            if (matcher === "89504e47") {
+                const schoolLogoImage = await pdfDoc.embedPng(bytes);
+                this.embedImage(pdfDoc, schoolLogoImage);
+            } else if (matcher.startsWith("ffd8")) {
+                const schoolLogoImage = await pdfDoc.embedJpg(bytes);
+                this.embedImage(pdfDoc, schoolLogoImage);
+            } else {
```

Comment:
This looks really ugly, and I'm not sure if it covers all possible cases. 

You should maybe use a library like https://www.npmjs.com/package/image-type

Or, if you don't want that, at least extract the image type check into a function.

Reply:
these are called magic numbers (the first bits of a file) and uniquely identify the type of the file, completely without knowing mimetype / extension. I don't want to introduce another lib here.